Buggyamb - ASP.NET Core debugging scenarios

ahmetmithat opened the pull request:
## Summary

BuggyAmb is just a very buggy ASP.NET Core Razor Pages based application. This application was intentionally created as a buggy application so it can be used as a learning resource to practice troubleshooting some sample problem scenarios:

* High CPU or low CPU performance problems, deadlocks.
* High memory issues.
* Crash issues and unexpected exceptions.

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/samples?pullRequest=4740) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/samples?pullRequest=4740) before we can accept your contribution.<br/><br/>:x: ahmetmithat [sign now](https://cla.dotnetfoundation.org/dotnet/samples?pullRequest=4740)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/samples?pullRequest=4740) it.</sub>

CamSoper reviewed (changes requested):
Overall, looks good. I'll provide suggestions.

CamSoper reviewed (changes requested):
Overall, looks great. Please remove the `.gitignore` and `.gitattributes` files, as those are handled at the repo level.

Since you're publishing an article elsewhere to cover the main content, I'm not too worried about doing an edit pass on the README. I think it's fine for a README, but I'd like @IEvangelist to give his opinion, too.

Awesome work, Ahmet!

IEvangelist commented on core/tutorials/buggyamb/Docs/README.md:
```diff
@@ -0,0 +1,7 @@
+# Installation instructions
```
In order for this sample to be discoverable we need to have the appropriate front matter. See https://review.docs.microsoft.com/en-us/help/contribute/samples/concepts/readme-template?branch=master

Also, I *think* this README should be at the root of the sample, so you'd have to move it down a directory - and update the refs to the other MD files. Here is a simple README with the minimal required front matter and content:

- https://github.com/dotnet/samples/blob/main/core/logging/custom-formatter/README.md
- https://docs.microsoft.com/en-us/samples/dotnet/samples/csharp-logging-fundamentals

IEvangelist reviewed (commented):
Need a bit of work done to the _README.md_, other than that - let's shipit.

ahmetmithat commented:
Hi Cam,

many thanks, I just removed the .gitignore and .gitattributes files.

Thanks,
AMB

From: Cam Soper ***@***.***>
Sent: Thursday, August 19, 2021 12:03 AM
To: dotnet/samples ***@***.***>
Cc: Ahmet Mithat Bostanci ***@***.***>; Author ***@***.***>
Subject: Re: [dotnet/samples] Buggyamb - ASP.NET Core debugging scenarios (#4740)


@CamSoper requested changes on this pull request.

Overall, looks great. Please remove the .gitignore and .gitattributes files, as those are handled at the repo level.

Since you're publishing an article elsewhere to cover the main content, I'm not too worried about doing an edit pass on the README. I think it's fine for a README, but I'd like @IEvangelist<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIEvangelist&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796698635%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ZYu7bzl%2BlJzdAWLn41RdnwD3kV1D6wJLoIgTaytcA9s%3D&reserved=0> to give his opinion, too.

Awesome work, Ahmet!

________________________________

In core/tutorials/buggyamb/.gitattributes<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23discussion_r691598209&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796708589%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=JjDxH8WuMMPqrXK4DnO4jVljCWlmyiaoCKas1X9DOm4%3D&reserved=0>:

> +###############################################################################

+# Set default behavior to automatically normalize line endings.

+###############################################################################

+* text=auto

+

+###############################################################################

+# Set default behavior for command prompt diff.

+#

+# This is need for earlier builds of msysgit that does not have it on by

+# default for csharp files.

+# Note: This is only used by command line

+###############################################################################

+#*.cs     diff=csharp

+

+###############################################################################

+# Set the merge driver for project and solution files

+#

+# Merging from the command prompt will add diff markers to the files if there

+# are conflicts (Merging from VS is not affected by the settings below, in VS

+# the diff markers are never inserted). Diff markers may cause the following

+# file extensions to fail to load in VS. An alternative would be to treat

+# these files as binary and thus will always conflict and require user

+# intervention with every merge. To do so, just uncomment the entries below

+###############################################################################

+#*.sln       merge=binary

+#*.csproj    merge=binary

+#*.vbproj    merge=binary

+#*.vcxproj   merge=binary

+#*.vcproj    merge=binary

+#*.dbproj    merge=binary

+#*.fsproj    merge=binary

+#*.lsproj    merge=binary

+#*.wixproj   merge=binary

+#*.modelproj merge=binary

+#*.sqlproj   merge=binary

+#*.wwaproj   merge=binary

+

+###############################################################################

+# behavior for image files

+#

+# image files are treated as binary by default.

+###############################################################################

+#*.jpg   binary

+#*.png   binary

+#*.gif   binary

+

+###############################################################################

+# diff behavior for common document formats

+#

+# Convert binary document formats to text before diffing them. This feature

+# is only available from the command line. Turn it on by uncommenting the

+# entries below.

+###############################################################################

+#*.doc   diff=astextplain

+#*.DOC   diff=astextplain

+#*.docx  diff=astextplain

+#*.DOCX  diff=astextplain

+#*.dot   diff=astextplain

+#*.DOT   diff=astextplain

+#*.pdf   diff=astextplain

+#*.PDF   diff=astextplain

+#*.rtf   diff=astextplain

+#*.RTF   diff=astextplain

You should delete this file. The .gitattributes file is set at the repo level.

________________________________

In core/tutorials/buggyamb/.gitignore<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23discussion_r691599249&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796708589%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=UlFhGoUOpZf5XQU3qjBLPDVOJlBStkx3WByY5NC4BZk%3D&reserved=0>:

> +_UpgradeReport_Files/

+Backup*/

+UpgradeLog*.XML

+UpgradeLog*.htm

+ServiceFabricBackup/

+*.rptproj.bak

+

+# SQL Server files

+*.mdf

+*.ldf

+*.ndf

+

+# Business Intelligence projects

+*.rdl.data

+*.bim.layout

+*.bim_*.settings

+*.rptproj.rsuser

+*- Backup*.rdl

+

+# Microsoft Fakes

+FakesAssemblies/

+

+# GhostDoc plugin setting file

+*.GhostDoc.xml

+

+# Node.js Tools for Visual Studio

+.ntvs_analysis.dat

+node_modules/

+

+# Visual Studio 6 build log

+*.plg

+

+# Visual Studio 6 workspace options file

+*.opt

+

+# Visual Studio 6 auto-generated workspace file (contains which files were open etc.)

+*.vbw

+

+# Visual Studio LightSwitch build output

+**/*.HTMLClient/GeneratedArtifacts

+**/*.DesktopClient/GeneratedArtifacts

+**/*.DesktopClient/ModelManifest.xml

+**/*.Server/GeneratedArtifacts

+**/*.Server/ModelManifest.xml

+_Pvt_Extensions

+

+# Paket dependency manager

+.paket/paket.exe

+paket-files/

+

+# FAKE - F# Make

+.fake/

+

+# JetBrains Rider

+.idea/

+*.sln.iml

+

+# CodeRush personal settings

+.cr/personal

+

+# Python Tools for Visual Studio (PTVS)

+__pycache__/

+*.pyc

+

+# Cake - Uncomment if you are using it

+# tools/**

+# !tools/packages.config

+

+# Tabs Studio

+*.tss

+

+# Telerik's JustMock configuration file

+*.jmconfig

+

+# BizTalk build output

+*.btp.cs

+*.btm.cs

+*.odx.cs

+*.xsd.cs

+

+# OpenCover UI analysis results

+OpenCover/

+

+# Azure Stream Analytics local run output

+ASALocalRun/

+

+# MSBuild Binary and Structured Log

+*.binlog

+

+# NVidia Nsight GPU debugger configuration file

+*.nvuser

+

+# MFractors (Xamarin productivity tool) working folder

+.mfractor/

+

+# Local History for Visual Studio

+.localhistory/

+

+# BeatPulse healthcheck temp database

+healthchecksdb

You should delete this file. The .gitignore file is set at the repo level.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23pullrequestreview-733344958&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796718550%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=JvyKaQme2pL%2B3TkqQksVW%2BNxGWazvSBoXV0ioKaCIiA%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAL2M37SCCBLSSMGPWUXOAVTT5QNXVANCNFSM5CMGGETA&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796718550%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=bKfhOoRIu0B6XpnjLVmrgJ1epHMrVQki%2FUH7MUC9taA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796728503%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ojq1OwzxluOImJ4Io8gldlFZ4Q55OpbyR0vwK7b8ew8%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796728503%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=byoL%2FuDaQ3Ny4av3Y4Son8MznDfJ8WNBVFPRLd93Mxc%3D&reserved=0>.

ahmetmithat commented:
Sorry, please ignore previous change, I removed wrong files but fixed it.

I have removed the correct .gitignore and .gitattributes files and pushed to the origin.

Thanks,
AMB

From: Ahmet Mithat Bostanci
Sent: Thursday, August 19, 2021 4:21 PM
To: dotnet/samples ***@***.***>; dotnet/samples ***@***.***>
Cc: Author ***@***.***>
Subject: RE: [dotnet/samples] Buggyamb - ASP.NET Core debugging scenarios (#4740)

Hi Cam,

many thanks, I just removed the .gitignore and .gitattributes files.

Thanks,
AMB

From: Cam Soper ***@***.******@***.***>>
Sent: Thursday, August 19, 2021 12:03 AM
To: dotnet/samples ***@***.******@***.***>>
Cc: Ahmet Mithat Bostanci ***@***.******@***.***>>; Author ***@***.******@***.***>>
Subject: Re: [dotnet/samples] Buggyamb - ASP.NET Core debugging scenarios (#4740)


@CamSoper requested changes on this pull request.

Overall, looks great. Please remove the .gitignore and .gitattributes files, as those are handled at the repo level.

Since you're publishing an article elsewhere to cover the main content, I'm not too worried about doing an edit pass on the README. I think it's fine for a README, but I'd like @IEvangelist<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FIEvangelist&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796698635%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ZYu7bzl%2BlJzdAWLn41RdnwD3kV1D6wJLoIgTaytcA9s%3D&reserved=0> to give his opinion, too.

Awesome work, Ahmet!

________________________________

In core/tutorials/buggyamb/.gitattributes<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23discussion_r691598209&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796708589%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=JjDxH8WuMMPqrXK4DnO4jVljCWlmyiaoCKas1X9DOm4%3D&reserved=0>:

> +###############################################################################

+# Set default behavior to automatically normalize line endings.

+###############################################################################

+* text=auto

+

+###############################################################################

+# Set default behavior for command prompt diff.

+#

+# This is need for earlier builds of msysgit that does not have it on by

+# default for csharp files.

+# Note: This is only used by command line

+###############################################################################

+#*.cs     diff=csharp

+

+###############################################################################

+# Set the merge driver for project and solution files

+#

+# Merging from the command prompt will add diff markers to the files if there

+# are conflicts (Merging from VS is not affected by the settings below, in VS

+# the diff markers are never inserted). Diff markers may cause the following

+# file extensions to fail to load in VS. An alternative would be to treat

+# these files as binary and thus will always conflict and require user

+# intervention with every merge. To do so, just uncomment the entries below

+###############################################################################

+#*.sln       merge=binary

+#*.csproj    merge=binary

+#*.vbproj    merge=binary

+#*.vcxproj   merge=binary

+#*.vcproj    merge=binary

+#*.dbproj    merge=binary

+#*.fsproj    merge=binary

+#*.lsproj    merge=binary

+#*.wixproj   merge=binary

+#*.modelproj merge=binary

+#*.sqlproj   merge=binary

+#*.wwaproj   merge=binary

+

+###############################################################################

+# behavior for image files

+#

+# image files are treated as binary by default.

+###############################################################################

+#*.jpg   binary

+#*.png   binary

+#*.gif   binary

+

+###############################################################################

+# diff behavior for common document formats

+#

+# Convert binary document formats to text before diffing them. This feature

+# is only available from the command line. Turn it on by uncommenting the

+# entries below.

+###############################################################################

+#*.doc   diff=astextplain

+#*.DOC   diff=astextplain

+#*.docx  diff=astextplain

+#*.DOCX  diff=astextplain

+#*.dot   diff=astextplain

+#*.DOT   diff=astextplain

+#*.pdf   diff=astextplain

+#*.PDF   diff=astextplain

+#*.rtf   diff=astextplain

+#*.RTF   diff=astextplain

You should delete this file. The .gitattributes file is set at the repo level.

________________________________

In core/tutorials/buggyamb/.gitignore<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23discussion_r691599249&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796708589%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=UlFhGoUOpZf5XQU3qjBLPDVOJlBStkx3WByY5NC4BZk%3D&reserved=0>:

> +_UpgradeReport_Files/

+Backup*/

+UpgradeLog*.XML

+UpgradeLog*.htm

+ServiceFabricBackup/

+*.rptproj.bak

+

+# SQL Server files

+*.mdf

+*.ldf

+*.ndf

+

+# Business Intelligence projects

+*.rdl.data

+*.bim.layout

+*.bim_*.settings

+*.rptproj.rsuser

+*- Backup*.rdl

+

+# Microsoft Fakes

+FakesAssemblies/

+

+# GhostDoc plugin setting file

+*.GhostDoc.xml

+

+# Node.js Tools for Visual Studio

+.ntvs_analysis.dat

+node_modules/

+

+# Visual Studio 6 build log

+*.plg

+

+# Visual Studio 6 workspace options file

+*.opt

+

+# Visual Studio 6 auto-generated workspace file (contains which files were open etc.)

+*.vbw

+

+# Visual Studio LightSwitch build output

+**/*.HTMLClient/GeneratedArtifacts

+**/*.DesktopClient/GeneratedArtifacts

+**/*.DesktopClient/ModelManifest.xml

+**/*.Server/GeneratedArtifacts

+**/*.Server/ModelManifest.xml

+_Pvt_Extensions

+

+# Paket dependency manager

+.paket/paket.exe

+paket-files/

+

+# FAKE - F# Make

+.fake/

+

+# JetBrains Rider

+.idea/

+*.sln.iml

+

+# CodeRush personal settings

+.cr/personal

+

+# Python Tools for Visual Studio (PTVS)

+__pycache__/

+*.pyc

+

+# Cake - Uncomment if you are using it

+# tools/**

+# !tools/packages.config

+

+# Tabs Studio

+*.tss

+

+# Telerik's JustMock configuration file

+*.jmconfig

+

+# BizTalk build output

+*.btp.cs

+*.btm.cs

+*.odx.cs

+*.xsd.cs

+

+# OpenCover UI analysis results

+OpenCover/

+

+# Azure Stream Analytics local run output

+ASALocalRun/

+

+# MSBuild Binary and Structured Log

+*.binlog

+

+# NVidia Nsight GPU debugger configuration file

+*.nvuser

+

+# MFractors (Xamarin productivity tool) working folder

+.mfractor/

+

+# Local History for Visual Studio

+.localhistory/

+

+# BeatPulse healthcheck temp database

+healthchecksdb

You should delete this file. The .gitignore file is set at the repo level.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fsamples%2Fpull%2F4740%23pullrequestreview-733344958&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796718550%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=JvyKaQme2pL%2B3TkqQksVW%2BNxGWazvSBoXV0ioKaCIiA%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAL2M37SCCBLSSMGPWUXOAVTT5QNXVANCNFSM5CMGGETA&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796718550%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=bKfhOoRIu0B6XpnjLVmrgJ1epHMrVQki%2FUH7MUC9taA%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796728503%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ojq1OwzxluOImJ4Io8gldlFZ4Q55OpbyR0vwK7b8ew8%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7CAhmet.Bostanci%40microsoft.com%7Ca7308eb2118f4b9c1a1908d9628b8a73%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637649173796728503%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=byoL%2FuDaQ3Ny4av3Y4Son8MznDfJ8WNBVFPRLd93Mxc%3D&reserved=0>.
